### PR TITLE
Fix a few migration issues with 2D weights

### PR DESCRIPTION
**Pull request by KiChjang**

### CODE REVIEW

File: pallets/parachain-system/src/migration.rs [hunk 1]
```diff
@@ -14,39 +14,53 @@
...
 			.saturating_add(T::DbWeight::get().writes(1));
 		StorageVersion::new(1).put::<Pallet<T>>();
 	}
 
 	weight
 }
 
-/// V1: `LastUpgrade` block number is removed from the storage since the upgrade
-/// mechanism now uses signals instead of block offsets.
-mod v1 {
```
Comment (KiChjang) [inline pallets/parachain-system/src/migration.rs:42]: I'm unsure if V1 has already enacted on chain, perhaps I shouldn't have replaced the code here?

File: pallets/xcmp-queue/src/migration.rs [hunk 2]
```diff
@@ -81,7 +82,10 @@ pub fn migrate_to_v2<T: Config>() -> Weight {
 			resume_threshold: pre.resume_threshold,
 			threshold_weight: Weight::from_ref_time(pre.threshold_weight),
 			weight_restrict_decay: Weight::from_ref_time(pre.weight_restrict_decay),
-			xcmp_max_individual_weight: Weight::from_ref_time(pre.xcmp_max_individual_weight),
+			xcmp_max_individual_weight: Weight::from_parts(
+				pre.xcmp_max_individual_weight,
+				MAX_POV_SIZE as u64,
```
Comment (bkchr) [inline pallets/xcmp-queue/src/migration.rs:87]: Not sure it is safe to use `MAX_POV_SIZE` everywhere?

File: pallets/parachain-system/src/migration.rs [hunk 3]
```diff
@@ -14,18 +14,19 @@
...
 /// The current storage version.
-pub const STORAGE_VERSION: StorageVersion = StorageVersion::new(1);
+pub const STORAGE_VERSION: StorageVersion = StorageVersion::new(2);
 
 /// Call this during the next runtime upgrade for this module.
 pub fn on_runtime_upgrade<T: Config>() -> Weight {
-	let mut weight: Weight = Weight::zero();
+	let mut weight: Weight = T::DbWeight::get().reads(2);
 
 	if StorageVersion::get::<Pallet<T>>() == 0 {
```
[the remark this answers is not in the document]
Comment (bkchr) [reply pallets/parachain-system/src/migration.rs:31]: Yeah, why not?

File: pallets/parachain-system/src/migration.rs [hunk 4]
```diff
@@ -34,9 +35,45 @@ pub fn on_runtime_upgrade<T: Config>() -> Weight {
...
+
 	weight
 }
 
+/// V2: Migrate to 2D weights for ReservedXcmpWeightOverride and ReservedDmpWeightOverride.
+mod v2 {
+	use super::*;
+
+	pub fn migrate<T: Config>() -> Weight {
+		let translate = |pre: u64| -> Weight { Weight::from_parts(pre, MAX_POV_SIZE as u64) };
```
Comment (gavofyork) [inline pallets/parachain-system/src/migration.rs:53]: do we really want to use `MAX_POV_SIZE` here? is it a limit?

Comment (KiChjang) [reply pallets/parachain-system/src/migration.rs:53]: It appears so:

https://github.com/paritytech/cumulus/blob/882a892b60e25449903052b946d25bd3a00f30eb/pallets/parachain-system/src/lib.rs#L804-L830

Used as the `max_weight` to `handle_dmp_messages`. It does not seem to be getting any of these numbers from the remaining block weights, so this is set arbitrarily by the developer.

Comment (bkchr) [reply pallets/parachain-system/src/migration.rs:53]: But we clearly don't want to use `MAX_POV_SIZE` as the max weight here. When we reach this point of executing the messages, we already don't have the `MAX_POV_SIZE` available anymore.

Comment (KiChjang) [reply pallets/parachain-system/src/migration.rs:53]: Okay, so let's give it... 64 KB?

Comment (bkchr) [reply pallets/parachain-system/src/migration.rs:53]: Sounds reasonable to me.

**Review by bkchr — APPROVED** [review]

**Review by shawntabrizi — APPROVED** [review]

**Review by ggwpez — APPROVED** [review]
Do messages already have a PoV attached? No, or?

**Pull request merged**